web/web.go

```diff
@@ -529,23 +531,52 @@ func signupCompleteOAuth(c *api.Context, w http.ResponseWriter, r *http.Request)
 			return
 		}
 
-		if result := <-api.Srv.Store.User().GetByAuth(team.Id, user.AuthData, service); result.Err == nil {
+		suchan := api.Srv.Store.User().GetByAuth(team.Id, user.AuthData, service)
+		euchan := api.Srv.Store.User().GetByEmail(team.Id, user.Email)
+
+		if team.Email == "" {
+			team.Email = user.Email
```

Comment:
You shouldn't be able to. The team is pulled from the DB and not supplied by the user/client.